packages/cozy-doctypes/src/File.js

```diff
@@ -148,6 +148,64 @@ class CozyFile extends Document {
       return `${filenameWithoutExtension}_1`
     }
   }
+
+  /**
+   * The goal of this method is to upload a file based on a conflict strategy.
+   * Be careful: We need to check if the file exists by doing a statByPath query
+   * before trying to upload the file since if we post and the stack return a
+   * 409 conflict, we will get a SPDY_ERROR_PROTOCOL on Chrome. This is the only
+   * viable workaround
+   * If there is no conflict, then we upload the file.
+   * If there is a conflict, then we applied the conflict strategy : `erase` or `rename`
+   * @param {String} name File Name
+   * @param {ArrayBuffer} file data
+   * @param {String} dirId dir id where to upload
+   * @param {String} onConflict Actually only 2 hardcoded strategies 'erase' or 'rename'
+   */
+  static async uploadFileWithConflictStrategy(name, file, dirId, onConflict) {
+    const filesCollection = this.cozyClient.collection('io.cozy.files')
+
+    try {
+      const path = await CozyFile.getFullpath(dirId, name)
+
+      const existingFile = await filesCollection.statByPath(path)
+      const { id: fileId } = existingFile.data
+      if (onConflict === 'erase') {
+        //!TODO Bug Fix. Seems we have to pass a name attribute ?!
+        const resp = await filesCollection.updateFile(file, {
+          dirId,
+          fileId,
+          name
+        })
+        return resp
+      } else {
```

Comment:
Shouldn't we use a `switch (onConflict)` that explicitly handles each possible case, and throw in the `default` case? Actually here we implicitly have a default value of `rename` for `onConflict`, since we explicitly handle the `erase` case, but apply the `rename` one on all other cases. It means if we pass `dont-exist` because we didn't read the doc or `earse` because we did a typo, it will use the `rename` strategy without any error.